raiden/exceptions.py

```diff
@@ -335,3 +335,14 @@ class MintFailed(RaidenError):
 
 class SerializationError(RaidenError):
     """ Invalid data are to be (de-)serialized. """
+
+
+class UnexpectedEthNodeError(RaidenUnrecoverableError):
+    """ The underlying eth node raised an unexpected error that has to be propagated. """
+
+    def __init__(self, from_exception: Exception):
+        super().__init__(f"Unexpected error in underlying eth node: {str(from_exception)}")
+
+
+class InsufficientEth(RaidenUnrecoverableError):
```

Comment:
Unrecoverable errors are exactly what they they're called... Unrecoverable. We should **not** try to catch them. Change this exception to recoverable.

Reply:
Right. I don't know why we were thinking of an unrecoverable error, while it was clear it would just result in a warning.

Reply:
Hmm but is it recoverable? If we don't have ETH then we are going to fail and crash in other unexpected ways.

Reply:
Thinking about it... you're actually right. It should be unrecoverable but i instantly thought that it should be recoverable because we catch the exception and log an error.

@jomuel sory i did not think this through. Let's just raise `RaidenUnrecoverable` and remove all handling of the exception. We should let it propagate and terminate the client.